src/test/java/com/google/jenkins/plugins/computeengine/integration/ComputeEngineCloud1WorkerCreatedFor2ExecutorsIT.java

```diff
@@ -0,0 +1,119 @@
+import hudson.slaves.NodeProvisioner.PlannedNode;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+import java.util.logging.SimpleFormatter;
+import java.util.logging.StreamHandler;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+import org.jvnet.hudson.test.JenkinsRule;
+
+public class ComputeEngineCloud1WorkerCreatedFor2ExecutorsIT {
+  private static Logger log =
+      Logger.getLogger(ComputeEngineCloud1WorkerCreatedFor2ExecutorsIT.class.getName());
+
+  private static final String MULTIPLE_NUM_EXECUTORS = "2";
+  private static final String EXPECTED_LOG_MESSAGE = "for excess workload of 2 units of label";
+
+  @ClassRule public static Timeout timeout = new Timeout(5, TimeUnit.MINUTES);
+  @ClassRule public static JenkinsRule jenkinsRule = new JenkinsRule();
+
+  private static ByteArrayOutputStream logOutput = new ByteArrayOutputStream();
+  private static StreamHandler streamHandler = new StreamHandler(logOutput, new SimpleFormatter());
+  private static ComputeClient client;
+  private static Map<String, String> label =
+      getLabel(ComputeEngineCloud1WorkerCreatedFor2ExecutorsIT.class);
+  private static Instance instance;
+
+  @BeforeClass
+  public static void init() throws Exception {
+    log.info("init");
+    initCredentials(jenkinsRule);
+    ComputeEngineCloud cloud = initCloud(jenkinsRule);
+    handleClassLogs(streamHandler, ComputeEngineCloud.class.getName());
+    client = initClient(jenkinsRule, label, log);
+    handleClassLogs(streamHandler, ComputeClient.class.getName());
+
+    cloud.addConfiguration(
+        instanceConfiguration(
+            DEB_JAVA_STARTUP_SCRIPT,
+            MULTIPLE_NUM_EXECUTORS,
+            LABEL,
+            label,
+            false,
+            false,
+            NULL_TEMPLATE));
+    // Add a new node
+    Collection<PlannedNode> planned = cloud.provision(new LabelAtom(LABEL), 2);
+
+    // There should be a planned node
+    assertEquals(logs(streamHandler, logOutput), 1, planned.size());
+
+    String name = planned.iterator().next().displayName;
+
+    // Wait for node creation to finish
+    planned.iterator().next().future.get();
+
+    instance = client.getInstance(PROJECT_ID, ZONE, name);
+  }
+
+  @AfterClass
+  public static void teardown() throws IOException {
+    ITUtil.teardown(streamHandler, logOutput, client, label, log);
+  }
+
+  @Test
+  public void test1WorkerCreatedFor2ExecutorsStatusRunning() {
+    assertEquals(logs(streamHandler, logOutput), "RUNNING", instance.getStatus());
+  }
+
+  @Test
+  public void test1WorkerCreatedFor2ExecutorsExpectedLogs() {
+    assertTrue(
+        logs(streamHandler, logOutput),
+        logs(streamHandler, logOutput).contains(EXPECTED_LOG_MESSAGE));
```

Comment:
EXPECTED_LOG_MESSAGE seems to only be used once here, I wonder if making it a constant is necessary, especially since an in-line string literal is used for the assertion in the test above.

Reply:
My general advice for constant vs. in-line literal is to defer to readability. In most cases there's a degree of readability inherent with string literals absent with most other data types.

Reply:
For me checking logs and based on that saying that test passed (if test is not about logging functionality) is useless..

Reply:
I don't know the original motivation for testing for the content of the logs, but I think it has a use in the context of integration tests. These have a broader scope than unit tests and test multiple interacting pieces of functionality. Testing that the user will see the correct log messages with a given configuration, especially logs that are generated in the code that is tested, is something that I think has value in that scope.

The broader scope here is external side effects, given that this should emulate the real deal rather than mocking that out. The state of the Jenkins instance (i.e. the Jenkins nodes), the gcp resources that are created and destroyed, and the logs that are generated are all side effects that could be relevant to these tests.

Reply:
From what I gather, I'm seconding what Stephen has said.
@evandbrown care to share your thoughts on the original motivation of including log testing in integration tests?